### PR TITLE
Fixes a runtime on destination tagger

### LINKED ISSUE #9817: Package wrap/destination tagger runtime
```
13:00:39 Runtime detected
list index out of bounds at packagewrap.dm:146
 proc name: attackby (/obj/item/delivery/attackby)
  source file: packagewrap.dm,146
  usr: George Memester (/mob/living/carbon/human)
  src: the large parcel (/obj/item/delivery/large)
  usr.loc: the floor (186,253,1) (/turf/simulated/floor)
  src.loc: the floor (186,254,1) (/turf/simulated/floor)
  call stack:
the large parcel (/obj/item/delivery/large): attackby(the destination tagger (/obj/item/device/destTagger), George Memester (/mob/living/carbon/human), "icon-x=14;icon-y=9;left=1;scre...")
George Memester (/mob/living/carbon/human): ClickOn(the large parcel (/obj/item/delivery/large), "icon-x=14;icon-y=9;left=1;scre...")
the large parcel (/obj/item/delivery/large): Click(the floor (186,254,1) (/turf/simulated/floor), "mapwindow.map", "icon-x=14;icon-y=9;left=1;scre...")

runtime error: list index out of bounds
proc name: attackby (/obj/item/delivery/attackby)
  source file: packagewrap.dm,146
  usr: George Memester (/mob/living/carbon/human)
  src: the large parcel (/obj/item/delivery/large)
  usr.loc: the floor (186,253,1) (/turf/simulated/floor)
  src.loc: the floor (186,254,1) (/turf/simulated/floor)
  call stack:
the large parcel (/obj/item/delivery/large): attackby(the destination tagger (/obj/item/device/destTagger), George Memester (/mob/living/carbon/human), "icon-x=14;icon-y=9;left=1;scre...")
George Memester (/mob/living/carbon/human): ClickOn(the large parcel (/obj/item/delivery/large), "icon-x=14;icon-y=9;left=1;scre...")
the large parcel (/obj/item/delivery/large): Click(the floor (186,254,1) (/turf/simulated/floor), "mapwindow.map", "icon-x=14;icon-y=9;left=1;scre...")
```

### PR DESCRIPTION
**Pull request by gbasood**
Fixes #9817

### CODE REVIEW

**Comment by gbasood** [comment]
OH WAIT I FUCKING KNOW WHY
BECAUSE THATS WHAT THE LABELLER LETS YOU DO

File: code/game/objects/items/stacks/packagewrap.dm [hunk 1]
```diff
@@ -196,6 +198,10 @@
 /obj/item/delivery/large/attack_hand(mob/user as mob)
 	if(wrapped)
 		wrapped.forceMove(get_turf(src.loc))
+	if(contents)
+		log_debug("[type] at [x],[y],[z] had [contents.len] contents after unwrapping the wrapped object")
```
Comment (ComicIronic) [inline code/game/objects/items/stacks/packagewrap.dm]: No sense in logging this for no reason.

**Comment by ghost** [comment]
Hello, can we get this in ASAP? Cargo is a pain right now.

File: code/game/objects/items/stacks/packagewrap.dm [hunk 2]
```diff
@@ -196,6 +198,10 @@
 /obj/item/delivery/large/attack_hand(mob/user as mob)
 	if(wrapped)
 		wrapped.forceMove(get_turf(src.loc))
+	if(contents.len)
+		log_debug("[type] at [x],[y],[z] had [contents.len] contents after unwrapping the wrapped object, ejecting contents")
+		for(var/atom/movable/A in contents)
+			A.forceMove(get_turf(src.loc))
```
Comment (PJB3005) [inline code/game/objects/items/stacks/packagewrap.dm]: Literally unmergable

Comment (Exxion) [reply code/game/objects/items/stacks/packagewrap.dm]: This line of code must have been written specifically to piss PJB off

Comment (PJB3005) [reply code/game/objects/items/stacks/packagewrap.dm]: Clearly.

File: code/game/objects/items/stacks/packagewrap.dm [hunk 3]
```diff
@@ -195,7 +197,11 @@
 
 /obj/item/delivery/large/attack_hand(mob/user as mob)
 	if(wrapped)
-		wrapped.forceMove(get_turf(src.loc))
+		wrapped.forceMove(get_turf(loc))
```
Comment (ComicIronic) [inline code/game/objects/items/stacks/packagewrap.dm]: Why not get_turf(src)?

File: code/modules/recycling/sortingmachinery.dm [hunk 4]
```diff
@@ -93,6 +93,7 @@
 	if(href_list["remove_dest"] && mode)
 		var/idx = Clamp(text2num(href_list["remove_dest"]), 1, destinations.len)
 		destinations -= destinations[idx]
+		currTag = 0 // In case the index was at the end of the list
```
Comment (ComicIronic) [inline code/modules/recycling/sortingmachinery.dm]: Don't do this, it's bad behaviour and I bet we'll get a bug report for it. Just do

```
if(currTag == destinations[idx])
   currTag = 0
```

before the destination is removed.

**Comment by SonixApache** [comment]
MERGE WHEN

**Comment by ComicIronic** [comment]
Never.

File: code/game/objects/items/stacks/packagewrap.dm (hunk 3 above)
Comment (gbasood) [reply code/game/objects/items/stacks/packagewrap.dm]: Skips a step upwards in the tree.
The `obj` has to at LEAST be in a `turf` and we already know it's not a turf, so it's just one step less.
Either that or the `obj` has a null `loc` and something bigger is fucking up

Comment (Exxion) [reply code/game/objects/items/stacks/packagewrap.dm]: `get_turf()` is basically equally fast regardless of nesting depth now
It does slow down marginally with deeper nesting, but it's only measurable on the scale of 10 million calls or more
It might even be slower to read `loc`

Comment (gbasood) [reply code/game/objects/items/stacks/packagewrap.dm]: [citation needed]

Comment (Exxion) [reply code/game/objects/items/stacks/packagewrap.dm]: #10877
TL;DR `get_turf()` now uses an internal BYOND function to get the turf and it's absurdly fast

**Comment by clusterfack** [comment]
Just fix the runtime only please, we've fixed the package wrap

**Comment by Exxion** [comment]
That debug output wouldn't have said anything you couldn't find out with VV anyway
Actually it wouldn't have said anything at all under nearly any circumstances but that's beside the point
